File: api/enums/roles.py
import graphene


class RoleEnums(graphene.Enum):
    USER_READ = "user_read"
    USER_WRITE = "user_write"
    ADMIN = "admin"
    SUPER_ADMIN = "super_admin"

    @property
    def description(self):
        if self == RoleEnums.USER_READ:
            return "A user who has been given access to view results"
        elif self == RoleEnums.USER_WRITE:
            return (
                "A user who has been given access to run scans, and " "manage domains"
            )
        elif self == RoleEnums.ADMIN:
            return (
                "A user who has the same access as a user write account, "
                "but can define new user write accounts"
            )
        elif self == RoleEnums.SUPER_ADMIN:
            return (
                "A user who has the same access as an admin, but can "
                "define new admins"
            )
        else:
            return "Another Role"

File: api/src/enums/roles.js
import { GraphQLEnumType } from 'graphql'

export const RoleEnums = new GraphQLEnumType({
  name: 'RoleEnums',
  values: {
    PENDING: {
      value: 'pending',
      description: 'A user who has requested an invite to an organization.',
    },
    USER: {
      value: 'user',
      description: 'A user who has been given access to view an organization.',
    },
    ADMIN: {
      value: 'admin',
      description:
        'A user who has the same access as a user write account, but can define new user read/write accounts.',
    },
    OWNER: {
      value: 'owner',
      description:
        'A user who has the same access as an admin, but can define new admins, and delete the organization.',
    },
    SUPER_ADMIN: {
      value: 'super_admin',
      description: 'A user who has the same access as an admin, but can define new admins.',
    },
    SERVICE: {
      value: 'service',
      description: 'An internal service used by Tracker to make changes to organizational data.',
    },
  },
  description: 'An enum used to assign, and test users roles.',
})
